From 9e580748fa04058767be93d03b662309d8f027da Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Tue, 11 Aug 2026 14:54:42 +0300
Subject: [PATCH] arm64: dts: lx2160a-clearfog-cx: enable qsfp port macs

The Clearfog-CX features a QSFP connector with zero support from
in-kernel drivers. For now four 10/25G macs are routed to it, and
partially enabled in device-tree.

Add managed = inband-status and enable the pcs nodes so that link state
reporting works for qsfp modules thta reqquire no runtime configuration.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../dts/freescale/fsl-lx2160a-clearfog-cx.dts | 20 +++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts b/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts
index 3970d16a2dc1..c481665e2cd6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-lx2160a-clearfog-cx.dts
@@ -19,21 +19,25 @@ / {
 &dpmac3 {
 	phys = <&serdes_1 7>, <&retimer0 3>, <&retimer1 3>;
 	phy-names = "serdes", "retimer", "retimer";
+	managed = "in-band-status";
 };
 
 &dpmac4 {
 	phys = <&serdes_1 6>, <&retimer0 2>, <&retimer1 2>;
 	phy-names = "serdes", "retimer", "retimer";
+	managed = "in-band-status";
 };
 
 &dpmac5 {
 	phys = <&serdes_1 5>, <&retimer0 1>, <&retimer1 1>;
 	phy-names = "serdes", "retimer", "retimer";
+	managed = "in-band-status";
 };
 
 &dpmac6 {
 	phys = <&serdes_1 4>, <&retimer0 0>, <&retimer1 0>;
 	phy-names = "serdes", "retimer", "retimer";
+	managed = "in-band-status";
 };
 
 &i2c_smb {
@@ -51,3 +55,19 @@ retimer1: retimer@23 {
 		#phy-cells = <1>;
 	};
 };
+
+&pcs_mdio3 {
+	status = "okay";
+};
+
+&pcs_mdio4 {
+	status = "okay";
+};
+
+&pcs_mdio5 {
+	status = "okay";
+};
+
+&pcs_mdio6 {
+	status = "okay";
+};
-- 
2.43.0

